arbor/mechcat.cpp

```diff
@@ -179,4 +230,14 @@ void mechanism_catalogue::copy_impl(const mechanism_catalogue& other) {
     }
 }
 
+void parameterize_over_ion(mechanism_catalogue& cat, const std::string& name, const std::string& ion) {
```

Comment:
A brief comment documenting the changes this function makes to the catalogue would help understanding.

Reply:
The comment in the header file is not enough? I'd prefer to expand that comment rather than add another here, if it is required.